File: open-graph/open-graph.php
<?php
declare(strict_types = 1);
extension_loaded('openssl') or die('openssl');

if ($argc != 2) {
   echo "open-graph.php <URL>\n";
   exit(1);
}

$url_s = $argv[1];
$get_s = file_get_contents($url_s);

$re_a = [
   '/ content="([^"]+)" property="og:image"/',
   '/="og:image" content="([^"]+)"/',
   '/="og:video" content="([^"]+)"/',
   '/=og:image content="([^"]+)"/'
];

foreach ($re_a as $re_s) {
   $mat_n = preg_match($re_s, $get_s, $get_a);
   if ($mat_n == 1) {
      echo $get_a[1], "\n";
   }
}

File: open-graph/open-graph.go
package main

import (
   "github.com/89z/x"
   "io"
   "log"
   "net/http"
   "os"
   "regexp"
)

func open(source string) (string, error) {
   x.LogInfo("Get", source)
   get, e := http.Get(source)
   if e != nil {
      return "", e
   }
   body, e := io.ReadAll(get.Body)
   if e != nil {
      return "", e
   }
   re := regexp.MustCompile(`="og:image" content="([^"]+)"`)
   return string(re.FindSubmatch(body)[1]), nil
}

func main() {
   if len(os.Args) != 2 {
      println("open-graph <URL>")
      os.Exit(1)
   }
   image, e := open(os.Args[1])
   if e != nil {
      log.Fatal(e)
   }
   println(image)
}
